docs/site/Express-middleware.md

```diff
@@ -0,0 +1,317 @@
+---
+lang: en
+title: 'Using Express middleware'
+keywords: LoopBack 4.0, LoopBack 4, Express, Middleware
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Express-middleware.html
+---
+
+## Overview
+
+Express is the most popular web framework for Node.js developers. As quoted
+below from
+[Express web site](https://expressjs.com/en/guide/using-middleware.html),
+middleware are the basic building blocks for Express applications.
+
+> Express is a routing and middleware web framework that has minimal
+> functionality of its own: An Express application is essentially a series of
+> middleware function calls.
+
+LookBack 4 leverages Express behind the scenes for its REST server
+implementation. We decided to not expose middleware capabilities to users while
+we pursue an elegant and non-invasive way to fit Express middleware into the
+LoopBack 4 programming model nicely. Meanwhile, we have received various
+requests and questions from our users on how to use Express middleware with
+LoopBack 4 or migrate their usage of Express middleware from LoopBack 3 to
+LoopBack 4.
+
+## Use cases
+
+The following use cases are identified to allow Express middleware to work with
+LoopBack 4:
+
+### For application developers
+
+1. Invoke one or more Express middleware handler functions in the REST sequence
+   before, after, or between existing actions.
+2. Apply Express middleware globally to all controllers, or locally to certain
+   controller classes and methods.
+3. Migrate LoopBack 3 applications that use Express middleware
+4. Be able to separate Express middleware configuration from its registration
+   and allow configuration changes at runtime to be effective without restarting
+   the application.
+
+### For extension developers
+
+1. Allow new actions to be added to the REST sequence without requiring code
+   changes to the application as long as the extension component is mounted to
+   the application. For example, `@loopback/authentication` contributes an
+   `Authenticate` action. It should be possible to build an extension module for
+   [`Helmet`](https://helmetjs.github.io/) to provide better protection for
+   LoopBack.
+2. Allow actions to be invoked in a configurable order.
+3. Allow extensions to leverage LoopBack's dependency injection and extension
+   point/extension capabilities.
+
+## Bring Express middleware to LoopBack
+
+Let's start with a few examples to illustrate how we can bring Express
+middleware to LoopBack applications with minimal efforts.
+
+### Use Express middleware within the sequence of actions
+
+Express middleware can now be plugged into the REST sequence with an
+`InvokeMiddleware` action being injected to the default sequence.
```

Comment:
> Express middleware can now be plugged into the REST sequence with an `InvokeMiddleware` action being injected to the default sequence.

Express middleware can now be plugged into the default REST sequence via the `InvokeMiddleware` action.

Reply:
Is this section the same as `#### Register Express middleware to be executed by `InvokeMiddleware` actions`?